File: LC_190.py
# 190. Reverse Bits

class Solution:
    def reverseBits(self, n: int) -> int:
        if n == 0:
            return 0
        
        result = 0
        for _ in range(32):
            result <<= 1
            result |= n & 1
            n >>= 1
        
        return result
    

File: LC_190.java
// 190. Reverse Bits

public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        if (n == 0)
            return 0;
        
        int result = 0;
        for (int i = 0; i < 32; i++) {
            result <<= 1;
            result |= n & 1;
            n >>= 1;
        }
        
        return result;
    }
}
